cherryontech/jupiter-jumpers
# Adjustments, settings page, favorites update

## Description

1. What does this PR do?
   - This PR implements the Favorites page and Settings page in the application.

2. List out what you did.
   - Updated the Favorites page component according to designs .
   - Developed the Settings page component with functionalities to change password and delete account.
   - Added styling for both pages to match the application’s theme.
   - Integrated the components into the main application routing.

## Related Issue

_Provide a link to the related issue in Jira._

https://cherryontech-jupiter-jumpers.atlassian.net/browse/SCRUM-62
https://cherryontech-jupiter-jumpers.atlassian.net/browse/SCRUM-64


- Feature update/change/addition

## Acceptance Criteria

_Include the checked-off acceptance criteria from the issue._
- [x] User can navigate to the Favorites page and see a list of favorite logs.
- [x] User can navigate to the Settings page and change their password.
- [x] User can navigate to the Settings page and delete their account.

## Update Screenshots

_Provide screenshots of the view before and after the changes._

### Before

(No Favorites and Settings pages)

### After

**Favorites Page:**
![Screenshot 2024-06-12 at 7 52 43 PM](https://github.com/cherryontech/Uncloud/assets/123599985/e551ee29-3b80-45eb-981a-eec77172ec62)

* still working on filters/ monthly update

**Settings Page:**
![Screenshot 2024-06-12 at 7 54 24 PM](https://github.com/cherryontech/Uncloud/assets/123599985/5b3c3c3b-f3a7-49b7-984c-0368efad7e98)



## Testing Instructions


1. Login and  navigate to the homepage.
2. Click on the "Favorites" link in the navigation bar to ensure the Favorites page loads correctly.
3. Verify that the Favorites page displays a list of favorite items.
4. Click on the "Settings" link in the navigation bar to ensure the Settings page loads correctly.
5. On the Settings page, test the Change Password functionality:
   - Enter the current password and a new password.
   - Verify that the password is successfully changed.
6. On the Settings page, test the Delete Account functionality:
   - Enter the current password.
   - Confirm account deletion and ensure the user is logged out and redirected to the logout confirmation page.
